submariner/IPSec-transport-mode.md

```diff
@@ -0,0 +1,106 @@
+
+# Cable Driver Enhancement - IPSec Transport Mode
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline the enhancements needed for the Submariner
+Cable Driver to enable IPSec in transport mode.
+
+For Overlay-over-IPSec in transport mode with further configuration, the benefits fall into the
+following categories:
+
+* **Performance**: Taking advantage of an overlay protocol wrapped in offloaded IPSec in transport mode can
+yield significant performance improvements. The existing IPSEC configuration performance on a pair of modern
+servers ~ 2Gbit maximum. Switching to IP GRE Tunnel protected by an offloaded transport mode IPSEC external
+“wrap” - 25Gbit. A difference of 12.5 times. Or in the case of VXLAN offloaded with an IPSec transport wrap - 9Gbit.
+* **Scale**: While it is theoretically possible to scale a system built around tunnel mode IPSEC to hundreds of nodes,
+it will be difficult to achieve reliability and stability, especially if there is a complex topology of tunnels. In
+order to achieve such scales, using other L2/L3 VPN tech combined with an optional external IPSEC protection provides
+a better approach, because it allows the use of dynamic routing protocols instead of complex static routing. Furthermore,
+it allows the use of traffic engineering and Site to Site QoS which is impossible in the case of IPSEC Tunnel mode.
+* **Debuggability**: Tunnel mode IPSEC is a debugging nightmare (especially if it is offloaded in part or in full). There
+is no visibility of what goes into the VPN and once it has entered the tunnel it is encrypted and thus unsuitable for debugging.
+This problem goes away if the VPN is using standard L2/L3 VPN technology which provides suitable interfaces on which the system
+administrator can snoop.
+
+## Proposal
+
+This enhancement proposes to extend the available IPSec offering in Submariner to also provide
```

Comment:
@mkolesnik Isn't it to scale Submariner's use cases to larger cluster sets at higher perf? The benefits above seem to kinda cover this.

Reply:
If it's a scale/perf issue then shouldn't we replace the default implementation rather than add another mode?

If, for our use cases, the two offer similar level of security but one is much slower than the other, why even keep the slow option?

I feel this proposal leaves much to the reader to understand the pros and cons, hence I asked for feedback from people more versed in IPSec.

Reply:
> If it's a scale/perf issue then shouldn't we replace the default implementation rather than add another mode?

We can do that in future once the implementation becomes stable and mature.

Reply:
I'm not saying we should immediately replace the current connectivity model with this, but if it suits our needs while offering better performance then I'd like to see a proposal to deprecate the current model.

I don't think we need to support two very similar tools for the same task, if one has a clear advantage over the other (as the EP suggests)